.gitignore

```diff
@@ -2,7 +2,28 @@
 *~
 .java-version
 .DS_Store
-.idea/
-.vscode
 /lib/
 target/
+
+# Gradle Files #
+################
+.gradle
+.m2
+.out/
+build
+dependencies.lock
+**/dependencies.lock
+gradle/wrapper/gradle-wrapper.jar
+
+# IDE specific files/directories #
+#######################################
```

Comment:
This is irrelevant to Gradle, isn't it?

Reply:
Despite not directly related - with plugin `idea` and `eclipse`, these files could be created after running init or build task. 

Would you suggest adding them with another PR as well ?